From 13904ed6c79d18dbb32673b2c2f1f95c48e4fadc Mon Sep 17 00:00:00 2001
From: Hiroshi Ohkubo <ohkubo.fs@gmail.com>
Date: Sat, 13 Feb 2016 14:48:22 +0900
Subject: [PATCH 1/3] Revert commit 9191d29

Revert "Set cdrom/suite and cdrom/codename
when premounted /cdrom found. Closes: #594344."

The bug will be fixed by the next commit,
as long as another one.

This reverts commit 9191d29c5b20c3216078fc9a4979a556cb2f7c89.

Signed-off-by: Hiroshi Ohkubo <ohkubo.fs@gmail.com>
---
 debian/cdrom-detect.postinst | 37 ++++++++++++++++---------------------
 1 file changed, 16 insertions(+), 21 deletions(-)

diff --git a/debian/cdrom-detect.postinst b/debian/cdrom-detect.postinst
index 6bf2454..da7dadf 100755
--- a/debian/cdrom-detect.postinst
+++ b/debian/cdrom-detect.postinst
@@ -39,24 +39,6 @@ try_mount() {
 	return $ret
 }
 
-# Set the suite and codename used by base-installer and base-config
-# to the suite/codename that is on the CD. In case there are multiple
-# suites, prefer the one in default-release.
-set_suite_and_codename() {
-	for dir in $(cat /etc/default-release) $(ls -1 /cdrom/dists/); do
-		relfile="/cdrom/dists/$dir/Release"
-		if [ -e $relfile ]; then
-			suite=$(sed -n 's/^Suite: *//p' "$relfile")
-			codename=$(sed -n 's/^Codename: *//p' "$relfile")
-			log "Detected CD with '$suite' ($codename) distribution"
-			db_set cdrom/suite "$suite"
-			db_set cdrom/codename "$codename"
-
-			break
-		fi
-	done
-}
-
 OS=$(udpkg --print-os)
 case "$OS" in
 	kfreebsd)
@@ -81,11 +63,10 @@ case "$OS" in
 esac
  
 # Is a cdrom already mounted? If so, assume it's the right one.
-mount | grep -q 'on /cdrom' && set_suite_and_codename && exit 0
+mount | grep -q 'on /cdrom' && exit 0
 if [ -e /cdrom/.disk/info ]; then
 	CDNAME=$(cat /cdrom/.disk/info)
 	log "Detected CD '$CDNAME'"
-	set_suite_and_codename
 	exit 0
 fi
 
@@ -253,7 +234,21 @@ for pooldir in /cdrom/pool/*/*; do
 done
 db_progress STOP
 
-set_suite_and_codename
+# Set the suite and codename used by base-installer and base-config
+# to the suite/codename that is on the CD. In case there are multiple
+# suites, prefer the one in default-release.
+for dir in $(cat /etc/default-release) $(ls -1 /cdrom/dists/); do
+	relfile="/cdrom/dists/$dir/Release"
+	if [ -e $relfile ]; then
+		suite=$(sed -n 's/^Suite: *//p' "$relfile")
+		codename=$(sed -n 's/^Codename: *//p' "$relfile")
+		log "Detected CD with '$suite' ($codename) distribution"
+		db_set cdrom/suite "$suite"
+		db_set cdrom/codename "$codename"
+
+		break
+	fi
+done
 
 if [ -z "$suite" ]; then
 	log "Error reading Release file; unable to determine distribution"
-- 
2.1.4

